tfx_addons/model_card_generator/executor_test.py

```diff
@@ -0,0 +1,204 @@
+           eval_artifacts=True,
+           example_stats_artifacts=True,
+           pushed_model_artifact=True,
+           exec_props=True),
+      dict(testcase_name='emptyInput',
+           eval_artifacts=False,
+           example_stats_artifacts=False,
+           pushed_model_artifact=False,
+           exec_props=False),
+      dict(testcase_name='partialInput',
+           eval_artifacts=False,
+           example_stats_artifacts=True,
+           pushed_model_artifact=False,
+           exec_props=True))
+  def test_do(self, eval_artifacts: bool, example_stats_artifacts: bool,
+              pushed_model_artifact: bool, exec_props: bool):
+
+    input_dict = {}
+    if eval_artifacts:
+      input_dict[standard_component_specs.EVALUATION_KEY] = self.eval_artifacts
+    if example_stats_artifacts:
+      input_dict[standard_component_specs.
+                 STATISTICS_KEY] = self.example_stats_artifacts
+    if pushed_model_artifact:
+      input_dict[standard_component_specs.PUSHED_MODEL_KEY] = [
+          self.pushed_model_artifact
+      ]
+
+    output_dict = {'model_card': [self.model_card_artifact]}
+
+    exec_properties = {}
+    if exec_props:
+      exec_properties['json'] = {
+          'model_details': {
+              'name': 'json_test',
+          }
+      }
+      exec_properties['template_io'] = [(self.template_file.full_path,
+                                         'my_cool_model_card.html')]
+
+    # Call MCT Executor
+    self.mct_executor.Do(input_dict=input_dict,
+                         output_dict=output_dict,
+                         exec_properties=exec_properties)
+
+    # Verify model card proto and document were generated
+    self.assertIn(
+        'model_card.proto',
+        os.listdir(os.path.join(self.model_card_artifact.uri, 'data')))
+    self.assertIn(
+        'default_template.html.jinja',
+        os.listdir(
+            os.path.join(self.model_card_artifact.uri, 'template', 'html')))
+
+    model_card_proto = model_card_pb2.ModelCard()
+    with open(
+        os.path.join(self.model_card_artifact.uri, 'data', 'model_card.proto'),
+        'rb') as f:
+      model_card_proto.ParseFromString(f.read())
+
+    with self.subTest(name='exec_props'):
+      model_card_dir = os.path.join(self.model_card_artifact.uri,
+                                    'model_cards')
+      if exec_props:
+        self.assertEqual(model_card_proto.model_details.name, 'json_test')
+        model_card_file_name = 'my_cool_model_card.html'
+      else:
+        model_card_file_name = 'model_card.html'
+      self.assertIn(model_card_file_name, os.listdir(model_card_dir))
+      model_card_filepath = os.path.join(model_card_dir, model_card_file_name)
+      with open(model_card_filepath) as f:
+        model_card_content = f.read()
+      if exec_props:
+        self.assertEqual(model_card_content, 'hello world')
+      else:
+        self.assertStartsWith(model_card_content, '<!DOCTYPE html>')
+
```

Comment:
With https://github.com/tensorflow/model-card-toolkit/pull/251 merged, we can remove  `confidence_interval=model_card_pb2.ConfidenceInterval()` and `sensitive=model_card_pb2.SensitiveData()` below. See [example diff](https://github.com/tensorflow/model-card-toolkit/pull/251/commits/a515b86dca5817e6172fa9994614410448c368d6#diff-f7f73c3eb6f562cbe2ddb2770da515990e31a0a145e1c8c377fea59e5955566c).

Reply:
Thank you for the hint! I removed the references.